File: 563-binary-tree-tilt.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int x) { val = x; }
 * }
 */
using System;
using System.Collections;
public class Solution {
    int tilt = 0;

    public int FindTilt(TreeNode root) {
        Helper(root);
        return tilt;
    }
    public int Helper(TreeNode root) {
        if (root == null)
            return 0;
        int left = Helper(root.left);
        int right = Helper(root.right);
        tilt += Math.Abs(left - right);
        return left + right + root.val;
    }
}

File: 563-binary-tree-tilt.go
package main

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func findTilt(root *TreeNode) int {
	res := 0
	var helper func(*TreeNode) int
	helper = func(node *TreeNode) int {
		if node == nil {
			return 0
		}
		left := helper(node.Left)
		right := helper(node.Right)
		res += abs(left - right)
		return left + right + node.Val
	}
	helper(root)
	return res
}

func abs(x int) int {
	if x < 0 {
		return -x
	}
	return x
}
